src/js/tasks-function.js

```diff
@@ -0,0 +1,181 @@
+        });
+      });
+      li[elem.index].addEventListener('mouseleave', (e) => {
+        this.updateTask(e.target.id, e.target.innerText);
+      });
+      inputBox[elem.index].addEventListener('change', (e) => {
+        const status = new TaskStatus();
+        if (e.target.checked === true) {
+          status.checked(this.taskData[elem.index]);
+        } else {
+          status.unchecked(this.taskData[elem.index]);
+        }
+        this.updateTask(
+          e.target.nextSibling.id,
+          e.target.nextSibling.innerText,
+        );
+      });
+      if (this.taskData[elem.index].completed === true) {
+        inputBox[elem.index].setAttribute('checked', 'checked');
+        li[elem.index].classList.add('checked');
+        label[elem.index].style.textDecoration = 'line-through';
+      } else if (this.taskData[elem.index].completed === false) {
+        inputBox[elem.index].removeAttribute('checked');
+        label[elem.index].style.textDecoration = 'none';
+      }
+    });
+  };
+
+  addtask = (description, completed, index) => {
+    const newtask = new Tasks(description, completed, index);
+    this.taskData.push(newtask);
+    if (typeof window !== 'undefined') {
+      localStorage.setItem('TodoListDB', JSON.stringify(this.taskData));
+    }
+  }
+
+  removetask = (item) => {
+    const key = item;
+    if (this.taskData.length === 1) {
+      this.taskData = [];
+    } else {
+      this.taskData.splice(key, 1);
+    }
+    this.taskData.forEach((elem, index) => {
+      elem.index = index;
+    });
+    if (typeof window !== 'undefined') {
+      localStorage.setItem('TodoListDB', JSON.stringify(this.taskData));
+      this.displayTask();
+    }
+  }
+
+  updateTask(item, description) {
+    if (this.taskData[item]) {
+      this.taskData[item].description = description;
+    }
+    this.taskData.forEach((elem, index) => {
+      elem.index = index;
+    });
+    if (typeof window !== 'undefined') {
+      localStorage.setItem('TodoListDB', JSON.stringify(this.taskData));
+      this.displayTask();
+    }
+  }
+
+  clearAllCompletTask = () => {
+    this.taskData = this.taskData.filter((elem) => elem.completed === false);
+    this.taskData.forEach((e, index) => {
+      e.index = index;
+    });
+    if (typeof window !== 'undefined') {
+      localStorage.setItem('TodoListDB', JSON.stringify(this.taskData));
+      window.location.reload()();
+    }
+  }
+}
+
+// Add New Task and Display it
+let index = 0;
+const completed = false;
+const entryTask = new Todolist();
+
+const newItemInput = document.querySelector('#new-item');
+const btnAddNewTask = document.querySelector('.btn-add');
+btnAddNewTask.addEventListener('click', () => {
+  if (newItemInput.value === '') {
+    btnAddNewTask.setCustomValidity('This is required field!');
+  } else {
+    entryTask.addtask(newItemInput.value, completed, index);
+    entryTask.displayTask();
+    newItemInput.value = '';
+    index += 1;
+  }
```

Comment:
- [OPTIONAL] I noticed that all tasks that I created have an `index` of `0` until I try to edit or delete a task before the `indexes` arrange to start from 0. I would advise you to refactor the index to start from `1` as per the requirement for this milestone and increase the index by one every time a task is added. Kindly see the gif I attached below to understand better👍 

![chrome-capture-2023-0-25](https://user-images.githubusercontent.com/77106962/214649963-f5237005-fe8c-4393-9e62-58f7c5a54f6d.gif)

Then when I try to edit, it rearranges

![chrome-capture-2023-0-25 (1)](https://user-images.githubusercontent.com/77106962/214650382-99d3cd23-cadf-4813-b17c-3f98e600c8f4.gif)